### PR TITLE
Add support for build.cmd -vs corehost.sln

### PR DESCRIPTION
**Pull request by mateoatr**
Should help with #50838.

### CODE REVIEW

File: eng/build.ps1 [hunk 1]
```diff
@@ -148,6 +148,25 @@ if ($vs) {
       }
     }
   }
+  elseif ($vs -ieq "corehost.sln") {
+    $archToOpen = $arch[0]
+    $configToOpen = $configuration[0]
+    if ($runtimeConfiguration) {
+      $configToOpen = $runtimeConfiguration
```
Comment (ViktorHofer) [inline eng/build.ps1:155]: Could you please avoid duplicating these variables?

**Review by vitek-karas — APPROVED** [review]
Except for @ViktorHofer 's comment this looks good.
It doesn't solve the main problem - it still builds everything (recompiles), CoreCLR.sln has the ability to just run the `-configureonly` which produces the projects but doesn't build anything.

But it's good to have this anyway.

**Comment by mateoatr** [comment]
> It doesn't solve the main problem - it still builds everything (recompiles), CoreCLR.sln has the ability to just run the -configureonly which produces the projects but doesn't build anything.

Yup. Will look into adding the `configureonly` part after merging this.

**Review by ViktorHofer — APPROVED** [review]

**Comment by ViktorHofer** [comment]
`runtime (Mono monointerpreter Pri0 Runtime Tests Run OSX x64 release)` failure is the xunit bug: https://github.com/dotnet/runtime/issues/11063

**Pull request merged**